src/librustc_codegen_ssa/back/link.rs

```diff
@@ -633,6 +644,44 @@ fn link_natively<'a, B: ArchiveBuilder<'a>>(
             continue;
         }
 
+        // Check to see if the link failed with "unrecognized command line option:
+        // '-static-pie'" for gcc or "unknown argument: '-static-pie'" for clang. If so,
+        // reperform the link step without the `-static-pie` but the `-static` option.
+        if (out.contains("unrecognized command-line option") || out.contains("unknown argument"))
+            && (out.contains("-static-pie") || out.contains("--no-dynamic-linker"))
+            && cmd.get_args().iter().any(|e| e.to_string_lossy() == "-static-pie")
+        {
+            info!("linker output: {:?}", out);
+            warn!("Linker does not support -static-pie command line option. Retrying without.");
+            let static_pie_crt = &sess
+                .target
+                .target
+                .options
+                .pre_link_objects_exe_crt_static_pie
+                .iter()
+                .map(|v| get_file_path(sess, v).into_os_string())
+                .collect::<Vec<_>>();
+            let mut static_crt_inserted = false;
+            for arg in cmd.take_args() {
+                if static_pie_crt.contains(&arg) {
+                    if !static_crt_inserted {
+                        for obj in &sess.target.target.options.pre_link_objects_exe_crt {
+                            cmd.arg(get_file_path(sess, obj));
+                        }
+                        static_crt_inserted = true;
+                    }
+                    continue;
+                }
+                if arg.to_string_lossy() != "-static-pie" {
+                    cmd.arg(arg);
+                } else {
+                    cmd.arg("-static");
+                }
+            }
+            info!("{:?}", &cmd);
```

Comment:
Can you at least add a prefix to explain what this line is. (e.g. `"new cmdline: {:?}"`)

Reply:
Sure, I just copied the block from the `-no-pie` handling above.